tao_compiler/mlir/disc/tests/regression/element_type_converter.cc

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/mlir/disc/tests/mlir_feature_test.h"
+#include "tensorflow/compiler/mlir/disc/tests/mlir_test.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace mlir_test {
+
+const std::string c_ft_path =
+    "tensorflow/compiler/mlir/disc/tests/regression/data/";
+
+TEST(ElementTypeConverterTest, Gemm) {
+  setenv("TAO_MLIR_ENABLE_AMP", "true", 1);
```

Comment:
This test actually cannot guarantee that AMP is enabled. Is there any way to check that?

Reply:
Or we can also rely on `[tao_compiler/mlir/disc/transforms/tests/element_type_converter.mlir](https://github.com/alibaba/BladeDISC/pull/308/files#diff-2bcd783bd2afe5ce7cd08a2cb6061dd0eb00ccf81c5d1b78ce569f019c0679d6)` for for the checking.

Reply:
File-check based UT?

Reply:
> File-check based UT?

File-check does not guarantee the cpp UT also works. Just like the checking of GPU stitch fusion. But AMP should be more robust than GPU stitch. The file-check based UT should be OK.